```
evaling implicit 'postInstall' string hook
panic: open /private/tmp/nix-build-nali-0.8.1.drv-0/Library/Application Support/nali/config.yaml: file exists

goroutine 1 [running]:
github.com/zu1k/nali/internal/config.ReadConfig({0xc00015e050, 0x48})
        github.com/zu1k/nali/internal/config/config.go:24 +0x6ed
main.main()
        github.com/zu1k/nali/main.go:13 +0x25
panic: open /private/tmp/nix-build-nali-0.8.1.drv-0/Library/Application Support/nali/config.yaml: file exists

goroutine 1 [running]:
github.com/zu1k/nali/internal/config.ReadConfig({0xc000028320, 0x48})
        github.com/zu1k/nali/internal/config/config.go:24 +0x6ed
main.main()
        github.com/zu1k/nali/main.go:13 +0x25
installShellCompletion: error: installed shell completion file `/nix/store/lqbknqwai5blmi8ywfqdcl1mgw4fka84-nali-0.8.1/share/bash-completion/completions/nali.bash' does not exist or has zero size
/nix/store/z6xrig32y1m1i3j7in3shya78y9agcp5-stdenv-darwin/setup: line 202: pop_var_context: head of shell_variables not a function context
```

--- a/pkgs/applications/networking/nali/default.nix
+++ b/pkgs/applications/networking/nali/default.nix
@@ -1,4 +1,4 @@
-{ lib, fetchFromGitHub, buildGoModule, installShellFiles }:
+{ lib, stdenv, fetchFromGitHub, buildGoModule, installShellFiles }:
 
 buildGoModule rec {
   pname = "nali";
@@ -19,11 +19,13 @@ buildGoModule rec {
   CGO_ENABLED = 0;
   ldflags = [ "-s" "-w" "-X github.com/zu1k/nali/internal/constant.Version=${version}" ];
 
-  postInstall = ''
-    installShellCompletion --cmd nali \
-      --bash <($out/bin/nali completion bash) \
-      --fish <($out/bin/nali completion fish) \
-      --zsh <($out/bin/nali completion zsh)
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+      export HOME="$TMPDIR"
+      # write to temp files to avoid race condition in config loading
+      $out/bin/nali completion bash > nali.bash
+      $out/bin/nali completion fish > nali.fish
+      $out/bin/nali completion zsh  > nali.zsh
+      installShellCompletion --cmd nali nali.{bash,fish,zsh}
   '';
 
   meta = with lib; {

